cmd/utils/flags.go

```diff
@@ -1945,16 +1947,40 @@ func MakeChain(ctx *cli.Context, stack *node.Node, readOnly bool) (chain *core.B
 	} else {
 		engine = ethash.NewFaker()
 		if !ctx.GlobalBool(FakePoWFlag.Name) {
+			datasetDir := stack.ResolvePath(eth.DefaultConfig.Ethash.DatasetDir)
+			ecip1099Block = config.GetEthashECIP1099Transition()
+			// check if ECIP-1099 is configured for this chain
+			if ecip1099Block != nil {
+				// ECIP-1099 is set, use etchash dir for DAGs instead
+				home := os.Getenv("HOME")
+				if home == "" {
+					if user, err := user.Current(); err == nil {
+						home = user.HomeDir
+					}
+				}
+				if runtime.GOOS == "darwin" {
+					datasetDir = filepath.Join(home, "Library", "Etchash")
+				} else if runtime.GOOS == "windows" {
+					localappdata := os.Getenv("LOCALAPPDATA")
+					if localappdata != "" {
+						datasetDir = filepath.Join(localappdata, "Etchash")
+					} else {
+						datasetDir = filepath.Join(home, "AppData", "Local", "Etchash")
+					}
+				} else {
+					datasetDir = filepath.Join(home, ".etchash")
+				}
+			}
```

Comment:
Here I left cache dir ethash as it exists withing the network scope anyway (~/.ethereum/classic/geth/ethash).
In terms of the datasetDir (where DAGs are kept for mining), i keep flipflopping on this, the idea is so miners that mine both ethash chains and etc, dont have conflicting DAGs, however a custom path can be set with a flag as is anyway in that situation. I'm not sure whether to leave this or take it out. Thoughts?

Reply:
I _believe_ the subcommands that use `MakeChain` will "inherit" the value of a flag `--ethash.dagdir` setting `eth.DefaultConfig.Ethash.DatasetDir` via `utils.SetEthConfig`. If so, I think we should keep the value that's set via that flag. If that flag isn't set (ie if `eth.DefaultConfig.Ethash.DatasetDir` hasn't been overidden by the user), then go ahead and use the different-namespaced default dir. 

Does that make sense?